Bearded.Utilities/Threading/ManualActionQueue.cs

```diff
@@ -85,96 +81,35 @@ public int ExecuteFor(TimeSpan time)
             return executed;
         }
 
-        #endregion
-
-        #region IActionQueue
-
-        /// <summary>
-        /// Queues an action to run. Returns immediately.
-        /// </summary>
-        /// <param name="action">The action to run.</param>
-        public void RunAndForget(Action action)
+        public void Queue(Action action)
         {
             actions.Add(action);
         }
 
-        /// <summary>
-        /// Queues an action to run. Returns only after the action has been executed.
-        /// </summary>
-        /// <param name="action">The action to run.</param>
-        public void RunAndAwait(Action action)
+        public Task Run(Action action)
         {
-            var reset = new ManualResetEvent(false);
+            var task = new TaskCompletionSource<object?>();
```

Comment:
nit: should this be Void?

Reply:
From what I've seen, using object and null is somewhat of a convention.